src/data/zmod/basic.lean

```diff
@@ -331,6 +331,38 @@ lemma cast_int_cast (k : ℤ) : ((k : zmod n) : R) = k :=
 
 end universal_property
 
+lemma int_coe_eq_int_coe_iff (a b : ℤ) (c : ℕ) :
+  (a : zmod c) = (b : zmod c) ↔ a ≡ b [ZMOD c] :=
+char_p.int_coe_eq_int_coe_iff (zmod c) c a b
+
+lemma nat_coe_eq_nat_coe_iff (a b c : ℕ) :
+  (a : zmod c) = (b : zmod c) ↔ a ≡ b [MOD c] :=
+begin
+  convert zmod.int_coe_eq_int_coe_iff a b c,
+  simp [nat.modeq.modeq_iff_dvd, int.modeq.modeq_iff_dvd],
+end
+
+@[simp]
+lemma int_coe_zmod_eq_zero_iff_dvd (a : ℤ) (b : ℕ) : (a : zmod b) = 0 ↔ (b : ℤ) ∣ a :=
+begin
+  change (a : zmod b) = ((0 : ℤ) : zmod b) ↔ (b : ℤ) ∣ a,
+  rw [zmod.int_coe_eq_int_coe_iff, int.modeq.modeq_zero_iff],
+end
+
+@[simp]
+lemma nat_coe_zmod_eq_zero_iff_dvd (a b : ℕ) : (a : zmod b) = 0 ↔ b ∣ a :=
+begin
+  change (a : zmod b) = ((0 : ℕ) : zmod b) ↔ b ∣ a,
+  rw [zmod.nat_coe_eq_nat_coe_iff, nat.modeq.modeq_zero_iff],
+end
+
+@[push_cast]
+lemma int_mod_coe_zmod_self (a : ℤ) (b : ℕ) : ((a % b : ℤ) : zmod b) = (a : zmod b) :=
```

Comment:
There's a version for `nat` called `cast_mod_nat`. I think the convention is for lemmas about `nat.cast` to have `cast` in the name.

Reply:
I swear some all of these lemmas existed before Johan's refactor.

Reply:
I've renamed this to `cast_mod_int`.